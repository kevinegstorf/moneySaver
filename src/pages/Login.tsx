import React, { FC } from "react";
import { Paper, Card, Button, TextField } from "@material-ui/core";
import { useHistory } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  card: {
    background: "linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)",
    borderRadius: 3,
    padding: 10,
  },
  paper: {
    maxWidth: "300px",
    margin: "0 auto",
    marginTop: "30vh",
  },
  form: {
    marginLeft: 5,
    display: "flex",
    flexDirection: "column",
  },
  buttonWrapper: {
    marginTop: 10,
  },
  button: {
    width: "100%",
  },
  input: {
    marginTop: 5,
  },
});

const Login: FC = () => {
  const [values, setValues] = React.useState({
    email: "",
    password: "",
    showPassword: false,
  });
  const classes = useStyles();

  const history = useHistory();

  const handleChange = (prop: any) => (event: any) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  const handleClick = () => {
    history.push("/home");
  };

  return (
    <Paper elevation={7} className={classes.paper}>
      <Card className={classes.card}>
        <form className={classes.form}>
          <TextField
            className={classes.input}
            id="email"
            type="email"
            value={values.email}
            autoFocus={true}
            onChange={handleChange("email")}
            label="Email Address"
            inputProps={{
              form: {
                autocomplete: "off",
              },
            }}
          />
          <TextField
            className={classes.input}
            id="password"
            color="secondary"
            type={values.showPassword ? "text" : "password"}
            value={values.password}
            autoComplete="off"
            onChange={handleChange("password")}
            label="Password"
          />
          <div className={classes.buttonWrapper}>
            <Button
              className={classes.button}
              id="submit"
              variant="outlined"
              color="primary"
              style={{ textTransform: "none" }}
              onClick={handleClick}
              size="large"
            >
              Login
            </Button>
          </div>
        </form>
      </Card>
    </Paper>
  );
};

export default Login;
