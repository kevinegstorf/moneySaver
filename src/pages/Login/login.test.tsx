import React from "react";
import Login from "./index";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent, cleanup } from "@testing-library/react";

// Mock useHistory
const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("login", () => {
  afterEach(cleanup);
  it("can fill in email an password", () => {
    // Arrange

    const emailText = "test@test.com";
    const passwordText = "password123";
    const { getByLabelText, getByText } = render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );

    const emailInput = getByLabelText("Email Address");
    const passwordInput = getByLabelText("Password");
    const button = getByText("Login");

    // Act
    fireEvent.change(emailInput, { target: { value: emailText } });
    fireEvent.change(passwordInput, { target: { value: passwordText } });
    fireEvent.click(button);

    // Assert
    expect(emailInput.value).toBe(emailText);
    expect(passwordInput.value).toBe(passwordText);
    expect(mockHistoryPush).toHaveBeenCalledWith("/home");
  });

  test("snapshot", () => {
    const { container } = render(<Login />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
