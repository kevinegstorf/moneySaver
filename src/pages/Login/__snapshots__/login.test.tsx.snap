// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login snapshot 1`] = `
<div
  class="MuiPaper-root makeStyles-paper-8 MuiPaper-elevation7 MuiPaper-rounded"
>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-7 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <form
      class="makeStyles-form-9"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-input-12"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-focused Mui-focused"
          data-shrink="true"
          for="email"
          id="email-label"
        >
          Email Address
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            form="[object Object]"
            id="email"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-input-12"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiInput-input"
            id="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="makeStyles-buttonWrapper-10"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-11 MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
          id="submit"
          style="text-transform: none;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
